nexus/src/app/sagas/finalize_disk.rs

```diff
@@ -287,19 +288,27 @@ async fn sfd_call_pantry_detach_for_disk(
     let params = sagactx.saga_params::<Params>()?;
     let pantry_address = sagactx.lookup::<SocketAddrV6>("pantry_address")?;
 
-    call_pantry_detach(
+    match call_pantry_detach(
         sagactx.user_data().nexus(),
         &log,
         params.disk_id,
         pantry_address,
     )
     .await
-    .map_err(|e| {
-        ActionError::action_failed(format!(
+    {
+        // If the detach succeeds, then proceed with finalization. If the detach
```

Comment:
If the pantry comes back later (like the sled was just being rebooted), what happens then?